### PR TITLE
`shorten-links` - Support new issue view

### PR DESCRIPTION
**Pull request by kovsu**
- For #7856 

## Test URLs

https://github.com/download-directory/download-directory.github.io/issues/141#issuecomment-2443099561

## Screenshot

| Before | After |
| ------ | ------ |
| image | image |
| image | image |

### CODE REVIEW

File: source/features/shorten-links.tsx [hunk 1]
```diff
@@ -5,7 +5,10 @@ import observe from '../helpers/selector-observer.js';
 
 /* This feature is currently so broad that it's not de-inited via signal, it's just run once for all pageloads #5889 */
 function init(): void {
-	observe(`.comment-body a[href]:not(.${linkifiedURLClass})`, shortenLink);
+	observe([
+		`.comment-body a[href]:not(.${linkifiedURLClass})`,
```
Comment (kovsu) [inline source/features/shorten-links.tsx:9]: I'm not sure if we can replace `comment-body` with `markdown-body` here.

Comment (fregante) [reply source/features/shorten-links.tsx:9]: It seems to me that we were avoiding to apply the feature to markdown comments, but I don't see any explicit decision to do that when blaming.

See live example: https://github.com/refined-github/sandbox/blob/default-a/renamed%20and%20edited.file.md

Let's keep it as is for now

File: source/github-helpers/dom-formatters.tsx [hunk 2]
```diff
@@ -20,10 +20,10 @@ export const codeElementsSelector = [
 ];
 
 export function shortenLink(link: HTMLAnchorElement): void {
-	// Exclude the link if the closest element found is not `.comment-body`
+	// Exclude the link if the closest element found is not `.markdown-body`
 	// This avoids shortening links in code and code suggestions, but still shortens them in review comments
 	// https://github.com/refined-github/refined-github/pull/4759#discussion_r702460890
-	if (link.closest(String([...codeElementsSelector, '.comment-body']))?.classList.contains('comment-body')) {
+	if (link.closest(String([...codeElementsSelector, '.markdown-body']))?.classList.contains('markdown-body')) {
```
Comment (fregante) [inline source/github-helpers/dom-formatters.tsx:26]: Make sure that this still works on the old version. See context in

https://github.com/refined-github/refined-github/pull/5472#issuecomment-1065227616

The link mentioned in the issue was added by https://chromewebstore.google.com/detail/octolinker/jlmafbaeoofdegohdhinkhilhclaklkp?hl=en&pli=1

Comment (kovsu) [reply source/github-helpers/dom-formatters.tsx:26]: > Make sure that this still works on the old version. See context in https://github.com/refined-github/refined-github/pull/5472#issuecomment-1065227616

It works here.

image

Comment (kovsu) [reply source/github-helpers/dom-formatters.tsx:26]: >The link mentioned in the issue was added by [chromewebstore.google.com/detail/octolinker/jlmafbaeoofdegohdhinkhilhclaklkp?hl=en&pli=1](https://chromewebstore.google.com/detail/octolinker/jlmafbaeoofdegohdhinkhilhclaklkp?hl=en&pli=1)

Here also works.

image

Comment (fregante) [reply source/github-helpers/dom-formatters.tsx:26]: Those links are not added by octolinker, they're already excluded by `linkifiedURLClass`

Comment (fregante) [reply source/github-helpers/dom-formatters.tsx:26]: > It works here.

This also doesn't contain any links, it was just an example of the current DOM structure.

Comment (fregante) [reply source/github-helpers/dom-formatters.tsx:26]: It seems that [OctoLinker is broken](https://togithub.com/OctoLinker/OctoLinker/issues/1907). I'm just testing this PR manually.

Comment (fregante) [inline source/github-helpers/dom-formatters.tsx:26]: I confirmed that by removing both this line _and_ `linkifiedURLClass` the links in https://github.com/refined-github/sandbox/pull/98 are (incorrectly) shortened, which means that this class works on the old version too.

I had broken `linkify-code` earlier which made it difficult to test.

- #7985

**Review by fregante — APPROVED** [review]

**Comment by fregante** [comment]
I found an unrelated bug in `shorten-links` 😬 

- This link is swallowed: https://github.com

Source:

```md
- This link is swallowed: https://github.com
```

This might be in https://github.com/refined-github/shorten-repo-url, can you check it out?

By the way, I sent you an invite :)

**Comment by fregante** [comment]
Welcome!

IMG_6348

**Comment by kidonng** [comment]
Welcome! 🤗 Albeit a bit late (and sorry for necro-bump)

**Pull request merged**